### PR TITLE
feat: JS compress and decompress lamports

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**
- tests
- instructions (`compress` and `decompress`)

merge only after #567

### CODE REVIEW

**Comment by SwenSchaeferjohann** [comment]
actually @ananas-block feel free to merge into #567 it's branched of from that and should make the CI for JS succeed.

File: js/stateless.js/tests/e2e/compress.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    let bob: Signer;
+
+    beforeAll(async () => {
+        rpc = await getTestRpc();
+        payer = await newAccountWithLamports(rpc);
+        bob = Keypair.generate();
+    });
+    // Note:
+    // We don't compress SOL yet, therefore cannot spend utxos with value yet.
+    // TODO: add one run with with inputUtxo where lamports: 0
```
Comment (ananas-block) [inline js/stateless.js/tests/e2e/compress.test.ts:28]: comments are not up to date right?

File: js/stateless.js/tests/e2e/compress.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            payer.publicKey,
+        );
+        const { blockhash: initBlockhash } = await rpc.getLatestBlockhash();
+        const signedInitTx = buildAndSignTx([ix], payer, initBlockhash);
+        await sendAndConfirmTx(rpc, signedInitTx);
+
+        const ixs = await LightSystemProgram.compress({
+            payer: payer.publicKey,
+            address: payer.publicKey,
+            lamports: 20,
```
Comment (ananas-block) [inline js/stateless.js/tests/e2e/compress.test.ts:40]: refactoring the lamports amount to a variable would make the test easier to maintain

File: js/stateless.js/tests/e2e/compress.test.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        await sendAndConfirmTx(rpc, signedDecompressTx);
+
+        //@ts-ignore
+        const indexedEvents2 = await rpc.getParsedEvents();
+        assert.equal(indexedEvents2.length, 3);
+        assert.equal(indexedEvents2[0].inputCompressedAccounts.length, 1);
+        assert.equal(indexedEvents2[0].outputCompressedAccounts.length, 1);
+        assert.equal(
+            Number(indexedEvents2[0].outputCompressedAccounts[0].lamports),
+            5,
```
Comment (ananas-block) [inline js/stateless.js/tests/e2e/compress.test.ts:116]: is there an assert that checks the recipient balance?

File: js/stateless.js/tests/e2e/compress.test.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            Number(indexedEvents[0].outputCompressedAccounts[0].lamports),
+            20,
+        );
+
+        assert.equal(
+            indexedEvents[0].outputCompressedAccounts[0].owner.toBase58(),
+            payer.publicKey.toBase58(),
+        );
+
+        assert.equal(indexedEvents[0].outputCompressedAccounts[0].data, null);
```
Comment (ananas-block) [inline js/stateless.js/tests/e2e/compress.test.ts:66]: is there an assert the checks that the sender balance has been reduced by the correct amount?

File: js/stateless.js/src/programs/compressed-pda.ts [hunk 5]
```diff
@@ -143,6 +225,32 @@ export class LightSystemProgram {
         return outputCompressedAccounts;
     }
 
+    static createDecompressOutputState(
+        inputCompressedAccounts: CompressedAccountWithMerkleContext[],
+        lamports: number | BN,
+    ): CompressedAccount[] {
+        lamports = bn(lamports);
+        const inputLamports = sumupLamports(inputCompressedAccounts);
```
Comment (ananas-block) [inline js/stateless.js/src/programs/compressed-pda.ts:233]: stumbled over the sumupLamports name, what about just sumLamports?

File: js/stateless.js/src/programs/compressed-pda.ts [hunk 6]
```diff
@@ -195,14 +303,203 @@ export class LightSystemProgram {
...
+        );
+
+        /// Format accounts
+        const staticAccounts = {
+            ...defaultStaticAccountsStruct(),
+            signer: payer,
+            invokingProgram: this.programId,
+            compressedSolPda: null, // this.deriveCompressedSolPda(),
+            deCompressRecipient: null,
+            systemProgram: null, //SystemProgram.programId,
```
Comment (ananas-block) [inline js/stateless.js/src/programs/compressed-pda.ts:318]: leftover comments?

File: js/stateless.js/src/programs/compressed-pda.ts [hunk 7]
```diff
@@ -195,14 +303,203 @@ export class LightSystemProgram {
...
         );
 
+        /// TODO : refactor
         /// Format accounts
         const staticAccounts = {
             ...defaultStaticAccountsStruct(),
             signer: payer,
             invokingProgram: this.programId,
+            compressedSolPda: this.deriveCompressedSolPda(),
+            deCompressRecipient: toAddress, // address, // TODO: confirm
```
Comment (ananas-block) [inline js/stateless.js/src/programs/compressed-pda.ts:501]: leftover comments?

File: js/stateless.js/src/programs/compressed-pda.ts (hunk 5 above)
Comment (SwenSchaeferjohann) [reply js/stateless.js/src/programs/compressed-pda.ts:233]: yea good catch i forgot to capitalize U. grammatically correct is sumUpLamports not sumupLamports nor sumLamports. fixed that.

File: js/stateless.js/tests/e2e/compress.test.ts (hunk 3 above)
Comment (SwenSchaeferjohann) [reply js/stateless.js/tests/e2e/compress.test.ts]: added

**Pull request merged**